src/btree/bt_prefetch.c

```diff
@@ -76,9 +76,27 @@ __wti_btree_prefetch(WT_SESSION_IMPL *session, WT_REF *ref)
          * these deleted pages into the cache if the fast truncate information is visible in the
          * session transaction snapshot.
          */
-        if (WT_REF_GET_STATE(next_ref) == WT_REF_DISK && F_ISSET(next_ref, WT_REF_FLAG_LEAF) &&
-          next_ref->page_del == NULL && !F_ISSET_ATOMIC_8(next_ref, WT_REF_FLAG_PREFETCH)) {
+        if (WT_REF_GET_STATE(next_ref) == WT_REF_DISK && next_ref->page_del == NULL &&
+          !F_ISSET_ATOMIC_8(next_ref, WT_REF_FLAG_PREFETCH)) {
+            WT_ASSERT(session, F_ISSET(next_ref, WT_REF_FLAG_LEAF));
```

Comment:
This assertion might be overkill, we should have the guarantee we only queue leaf pages.

Reply:
Pushing for this as d81eb193acdc6397bd1f6d215d8faf4fefad2a1b reverted the leaf check.

Reply:
done